src/stats.c

```diff
@@ -1127,10 +1130,15 @@ void ClearError ( void )
         }
 #ifdef USE_GASMAN
         /* and check if maximal memory was overrun */
-        if ( SyStorOverrun != 0 ) {
-          SyStorOverrun = 0; /* reset */
-          Pr("GAP has exceeded the permitted memory (-o option),\n", 0, 0);
-          Pr("the maximum is now enlarged to %d kB.\n", (Int)SyStorMax, 0);
+        if (SyStorOverrun != SY_STOR_OVERRUN_CLEAR) {
+            SyStorEnum value = SyStorOverrun;
+            SyStorOverrun = SY_STOR_OVERRUN_CLEAR; /* reset */
+            if (value == SY_STOR_OVERRUN_TO_REPORT) {
+                Pr("GAP has exceeded the permitted memory (-o option),\n", 0,
+                   0);
+                Pr("the maximum is now enlarged to %d kB.\n", (Int)SyStorMax,
+                   0);
+            }
```

Comment:
Or, just do this:
```C
if (SyStorOverrun == SY_STOR_OVERRUN_TO_REPORT) {
  Pr(...);
}
SyStorOverrun = SY_STOR_OVERRUN_CLEAR;
```

Reply:
My reason for doing it the way I've done it was because I was worried about calling other functions before I'd set SyStorOverrun. However, Pr doesn't call into GAP code, so that shouldn't be a problem.